### ISSUE TITLE
USWDS - Documentation: Allow sidenav reorder via settings

### ISSUE THREAD

**Issue by mejiaj** [uswds/uswds#5807]
## Description

Improve how we address breaking changes. 

Give users the opportunity to fallback to previous behavior, but recommend and show _how_ to implement the new behavior. 

- [ ] Create setting for sidenav reorder
  - [x] Default is `FALSE` for new behavior
  - [x] Setting to `TRUE` for legacy implementations will revert to previous display and give warning.
  - [ ] Add SASS notification
- [ ] Update guidance
  - [ ] Update component guidance
  - [x] Update settings page on site
  - [x] Create changelog entry (for component & settings page)

_From [slack discussion 🔒 ](https://gsa-tts.slack.com/archives/C050HRGN7/p1709832741477039?thread_ts=1708701466.669389&cid=C050HRGN7) in `#uswds`_ 

## Additional information [optional]

Extension of work done in #4594.
